odl/trafos/wavelet.py

```diff
@@ -750,11 +828,26 @@ def __init__(self, ran, nscales, wbasis, mode):
 
             'per': periodization -- like periodic-padding but gives the
             smallest possible number of decomposition coefficients.
+
+            With J.-O. Stomberg's biorthogonal wavelets only signal
+            extension mode 'sym' is available.
         """
         self.nscales = int(nscales)
-        self.wbasis = wbasis
         self.mode = str(mode).lower()
 
+        if isinstance(wbasis, pywt.Wavelet):
+            self.wbasis = wbasis
+        else:
+            try:
+                wbasis = wbasis + ''
+                if wbasis.startswith('JOS'):
+                    self.wbasis = wbasis
```

Comment:
The simplest possibility, and I guess that's what @adler-j means, is to always expect a string and only differentiate between `wbasis.startswith('jos')` (lowercase preferred). For the pywavelets strings, you would use the string to initialize a `Wavelet` object, otherwise you call the corresponding function in Jan-Olov's library.

In the long run, I suggest to use a flag like `'backend'` with choices `pywavelets` or `jos` and separate the wavelet families according to that. Then add a property `backend` to the classes, that makes checking for implementation easy. You can also split the `_call` code accordingly, i.e. make a `_call_pyfftw` and a `_call_jos` and just call those from the `_call` depending on backend.